File: 프로그래머스/3/42861. 섬 연결하기/섬 연결하기.js
function solution(n, costs) {
    const parent = new Array(n).fill(0).map((_, idx) => idx);
    
    function find(x){
        if(parent[x] === x) return x;
        return parent[x] = find(parent[x]);
    }
    
    function union(a, b) {
        const rootA = find(a);
        const rootB = find(b);
        
        if(rootA !== rootB) parent[rootB] = rootA;
    }
    
    costs.sort((a, b) => a[2] - b[2]);
    
    let result = 0;
    for(const [a, b, cost] of costs){
        if(find(a) !== find(b)){
            union(a, b);
            result += cost;
        }
    }
    return result;
}

File: 프로그래머스/3/42861. 섬 연결하기/섬 연결하기.py
def solution(n, costs):
    parent = [i for i in range(n)]
    
    def find(x):
        if parent[x] == x:
            return x
        return find(parent[x])
        
    def union(a, b):
        rootA = find(a)
        rootB = find(b)
        
        if rootA != rootB:
            parent[rootB] = rootA
        
    costs.sort(key = lambda x: x[2])
    
    answer = 0
    for a, b, cost in costs:
        if find(a) != find(b):
            union(a, b)
            answer += cost
    return answer